Source/quests.cpp

```diff
@@ -264,54 +264,38 @@ void PrintQLString(const Surface &out, int x, int line, const char *str)
 
 void InitQuests()
 {
-	if (!gbIsMultiplayer) {
-		for (auto &quest : Quests) {
-			quest._qactive = QUEST_NOTAVAIL;
-		}
-	} else {
-		int q = 0;
-		for (auto &quest : Quests) {
-			if (QuestData[q].isSinglePlayerOnly) {
-				quest._qactive = QUEST_NOTAVAIL;
-			}
-			q++;
-		}
-	}
-
 	QuestDialogTable[TOWN_HEALER][Q_MUSHROOM] = TEXT_NONE;
 	QuestDialogTable[TOWN_WITCH][Q_MUSHROOM] = TEXT_MUSH9;
 
 	QuestLogIsOpen = false;
 	WaterDone = 0;
-	int initiatedQuests = 0;
 
+	int initiatedQuests = 0;
 	int q = 0;
 	for (auto &quest : Quests) {
-		if (!gbIsMultiplayer || !QuestData[q].isSinglePlayerOnly) {
-			quest._qtype = QuestData[q]._qdtype;
-			if (gbIsMultiplayer) {
-				quest._qlevel = QuestData[q]._qdmultlvl;
-				if (!delta_quest_inited(initiatedQuests)) {
-					quest._qactive = QUEST_INIT;
-					quest._qvar1 = 0;
-					quest._qlog = false;
-				}
-				initiatedQuests++;
-			} else {
+		quest._qidx = static_cast<quest_id>(q);
+		auto &questData = QuestData[q];
+		q++;
+
+		quest._qactive = QUEST_NOTAVAIL;
+		quest.position = { 0, 0 };
+		quest._qlvltype = questData._qlvlt;
+		quest._qslvl = questData._qslvl;
+		quest._qvar1 = 0;
+		quest._qvar2 = 0;
+		quest._qlog = false;
+		quest._qmsg = questData._qdmsg;
+
+		if (!gbIsMultiplayer) {
+			quest._qlevel = questData._qdlvl;
+			quest._qactive = QUEST_INIT;
+		} else if (!questData.isSinglePlayerOnly) {
+			quest._qlevel = questData._qdmultlvl;

```

Comment:
Just noticed this in the original code, the only time this does anything useful is for Q_DIABLO where it sets qlevel to -1 instead of 15. Every other case is a no-op since `qdmultlvl == qlevel` for every other case when `isSinglePlayerOnly == false`. Is that a pre-existing bug?

Reply:
I don't understand that to be a bug, just not very interesting given the quests in the original game

Reply:
I see, the drlg_l4 checks involving Q_DIABLO never look at qlevel so this value isn't used for that quest.